pkg/providers/amifamily/ami.go

```diff
@@ -94,69 +94,86 @@ func (p *Provider) KubeServerVersion(ctx context.Context) (string, error) {
 	return version, nil
 }
 
-// Get returns a set of AMIIDs and corresponding instance types. AMI may vary due to architecture, accelerator, etc
+// MapInstanceTypes returns a set of AMIIDs and corresponding instance types. AMI may vary due to architecture, accelerator, etc
 // If AMI overrides are specified in the AWSNodeTemplate, then only those AMIs will be chosen.
-func (p *Provider) Get(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate, instanceTypes []*cloudprovider.InstanceType, amiFamily AMIFamily) (map[string][]*cloudprovider.InstanceType, error) {
-	kubernetesVersion, err := p.KubeServerVersion(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("getting kubernetes version %w", err)
-	}
+func (p *Provider) MapInstanceTypes(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate, instanceTypes []*cloudprovider.InstanceType, options *Options) (map[string][]*cloudprovider.InstanceType, error) {
```

Comment:
I think we might be able to collapse some of this. If the requirements returned from `Get` is a slice and is already pre-sorted by creation date and name, then we can use this as the AMI data type

```go
type AMI struct {
	Name         string
	AmiID        string
	CreationDate string
	Requirements []v1.NodeSelectorRequirement
}
```

and we can just have a simple function that assigns the AMI to the instance type

```go
func mapInstanceTypes(amis []AMI, instanceTypes []*cloudprovider.InstanceType) (res map[string][]*cloudprovider.InstanceType) {
	for _, instanceType := range instanceTypes {
		for _, ami := range amis {
			if err := instanceType.Requirements.Compatible(scheduling.NewNodeSelectorRequirements(ami.Requirements...)); err == nil {
				res[ami.AmiID] = append(res[ami.AmiID], instanceType)
				break
			}
		}
	}
}
```